### PR TITLE
Minor fixes.

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

Mostly stuff to tidy up the static methods in the CB.

- Remove default pattern from `parse_to_table` (caused IDE to freeze).
- Rename any `_` arguments to what they are.
- Merge `Date.now` into `Date.today`
- Merge the Interval constructors into a single constructor.
- Hide various methods.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso [hunk 1]
```diff
@@ -264,7 +265,8 @@ type Any
 
              "Hello".if_nothing ""
     if_nothing : Any -> Any
-    if_nothing self ~_ = self
+    if_nothing self ~other =
+        const self other
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Any.enso:269]: Feel free to leave as is, but reading this code I actually feel like including the logic for ignoring the value in-line could make this more readable:

```suggestion
    if_nothing self ~other =
        ignore_input ~_=Nothing
        ignore_input other
        self
```

but I guess `const` is all good to as long as you know its definition suspends the argument

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso [hunk 2]
```diff
@@ -48,12 +48,15 @@ type Json
             parsed = json_parse json
             make_enso parsed
 
-    ## Serialize an Object to JSON
+    ## PRIVATE
+       ADVANCED
+       Serialize an Object to JSON
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso]:
```suggestion
       Serialize an Object to JSON.
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering.enso [hunk 3]
```diff
@@ -161,7 +161,9 @@ type Ordering
     ## A representation that the first value orders as greater than the second.
     Greater
 
-    ## Compares to values and returns an Ordering
+    ## PRIVATE
+       ADVANCED
+       Compares to values and returns an Ordering
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering.enso]:
```suggestion
       Compares values and returns an Ordering.
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso [hunk 4]
```diff
@@ -18,7 +18,9 @@ type Set
     ## PRIVATE
     Value (underlying_map : Map Any Nothing)
 
-    ## Constructs a new set from a vector.
+    ## PRIVATE
+       ADVANCED
+       Constructs a new set from a vector.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso:23]: Why are we making the Set hidden in CB? Isn't it useful to sometimes work with sets?

**Review by radeusgd — APPROVED** [review]
Looks good, just some minor comments

File: distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso [hunk 5]
```diff
@@ -88,7 +88,10 @@ flip f = (x -> y -> f y x)
    > Example
     IO.println <| [1, 2, 3].map (Function.const 7) # Prints '[7, 7, 7]'
 const : Any -> Any -> Any
-const x _ = x
+const x ~f =
+    black_hole ~_ = Nothing
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso:92]: Maybe we should have a private helper method for this defined somewhere. `mark_used` or something like that.

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso (hunk 4 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso:23]: At the moment we don't have APIs using it, so decided to hide it for now. I am also looking over the `from` methods to replace with `to` extensions. Hopefully will get that together tomorrow.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso [hunk 6]
```diff
@@ -34,7 +36,9 @@ type Array
     empty : Array
     empty = @Builtin_Method "Array.empty"
 
-    ## Creates a new array of length size, with all elements uninitialized.
+    ## PRIVATE
+       ADVANCED
+       Creates a new array of length size, with all elements uninitialized.
```
Comment (JaroslavTulach) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso:41]: Is this actually useful for anything? `Array` is immutable these days - who can use uninitialized array of some size?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso (hunk 5 above)
Comment (JaroslavTulach) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso:92]: Shouldn't we rather get `~_` working?

**Review by JaroslavTulach — APPROVED** [review]
`EnsoDate` changes are certainly fine.

**Review by Akirathan — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso (hunk 5 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso:92]: The problem is that `_` argument appear as Internal in the IDE so any public function should have named arguments.
The `_` (and `~_`) do work internally or for PRIVATE functions.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso (hunk 6 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso:41]: Fair point will see if we use it anywhere anymore and remove if not in next tidy up PR.

**Pull request merged**